Gems/Atom/RHI/Code/Source/RHI.Reflect/PipelineLayoutDescriptor.cpp

```diff
@@ -18,15 +18,18 @@ namespace AZ
             if (SerializeContext* serializeContext = azrtti_cast<SerializeContext*>(context))
             {
                 serializeContext->Class<ResourceBindingInfo>()
-                    ->Version(0)
+                    ->Version(1)
                     ->Field("m_shaderStageMask", &ResourceBindingInfo::m_shaderStageMask)
-                    ->Field("m_registerId", &ResourceBindingInfo::m_registerId);
+                    ->Field("m_registerId", &ResourceBindingInfo::m_registerId)
+                    ->Field("m_spaceId", &ResourceBindingInfo::m_spaceId);
             }
         }
 
         HashValue64 ResourceBindingInfo::GetHash() const
         {
-            return TypeHash64(*this);
+            HashValue64 hash = TypeHash64(static_cast<uint32_t>(m_shaderStageMask));
+            hash = TypeHash64(m_registerId, hash);
+            return hash;
```

Comment:
Yea we don't actually. This gotcha is explained in a different comment next to the member declaration. Basically, the issue is that the material compiler uses the hash to check for SRG layout compatibility, but we actually don't want the space id to be included in that check. We probably need to fix this in the material compiler at some point, but since material canvas work will be changing how materials are built later, I left this for another time.